notes/scope format.md

```diff
@@ -0,0 +1,48 @@
+## Some notes
+
+Make sure that the begin and end are relative to the start of the line
+
+You _generally_ should not have multiple of the same scopes inside one another (directly)
+
+## Rules
+
+-   Arguments should have the surrounding scope `["argument", "parser name"]`.
+    -   Example: `summon ~ ~ ~ zombie`
+
+```ts
+{
+    end: 7,
+    scopes: ["argument", "minecraft:block_pos"],
+    start: 12
+}
+```
+
+-   Scopes of multiple names should be separated by `-`
+    -   Example: in `say @e[name="foo",tag=bar]` there should be
+
+```ts
+{
+    end: 18,
+    scopes: ["kvpair-separator", "separator"],
+    start: 17
+}
+```
+
+## Various scopes
+
```

Comment:
The number of scopes here seems very complicated. You might also want to see Microsoft/language-server-protocol#124. Having read through the arguments there, I have realised that it's probably better if the server sends highlighting, because we can then send it for each line (This is only better for our server)

Reply:
Additionally, we might want to only support a more simple highlighting method initially until https://github.com/Microsoft/language-server-protocol/issues/18 is resolved, which seems to be proceeding in https://github.com/Microsoft/vscode-languageserver-node/pull/367. We don't want to make a massively complex system which becomes deprecated in a few weeks.